### PR TITLE
DDF-4320 Fix CDM ingesting too soon on system start

### PR DESCRIPTION
**Pull request by peterhuffer**
##### ABBREVIATED REVIEW BETWEEN 2.13.X AND MASTER IS IN EFFECT
Link to 2.13.x PR: #4086 
There were some minor merge conflicts and a few dependency changes due to #3842. Should be retested according to 2.13.x PR steps.

> Set log:set TRACE `org.codice.ddf.catalog.transformer.bootflag` for detailed logs.
> Verify the following scenarios.
> 
> * Start a normal CDM (move, delete, or inplace) and verify it begins ingesting
> * Start a normal CDM (move, delete, or inplace). Restart the distribution and verify it does not start until all InputTransformers are available (view logs).
> * Before a fresh install, edit the ddf-default-input-transformers.json file to include an InputTransformer ID that doesn't exist. Verify install fails.
> 

#### Who is reviewing it? 
@emmberk @kcover @paouelle 

#### Select relevant component teams: 
@codice/io 

#### Ask 2 committers to review/merge the PR and tag them here.
@clockard
@tbatie

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/core/catalog-core-directorymonitor/pom.xml [hunk 1]
```diff
@@ -361,12 +385,7 @@
                             ddf-security-common,
                             platform-util,
                             platform-util-unavailableurls,
-                            failsafe,
-                            hazelcast,
-                            sardine,
-                            httpclient,
-                            httpcore,
-                            commons-lang3
+                            sardine
```
Comment (bakejeyner) [inline catalog/core/catalog-core-directorymonitor/pom.xml]: ❓ Were these embeds not needed anymore?

Comment (peterhuffer) [reply catalog/core/catalog-core-directorymonitor/pom.xml]: Yes. The bundles are now started as part of the feature instead.

File: catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitor.java [hunk 2]
```diff
@@ -95,33 +93,15 @@
...
    * Constructs a monitor that uses the given RetryPolicy while waiting for the content scheme, and
    * the given Executor to run the setup and Camel configuration.
    *
    * @param camelContext the Camel context to use across all Content directory monitors.
+   * @param attributeRegistry {@link AttributeRegistry} to use to perform attribute lookup for
+   *     attribute overrides for in-place monitoring
    * @param maxRetries Policy for polling the 'content' CamelComponent. Specifies, for any content
    *     directory monitor, the number of times it will poll.
-   * @param delayBetweenRetries Policy for polling the 'content' CamelComponent. Specifies, for any
-   *     content directory monitor, the number of seconds it will wait between consecutive polls.
```
[the remark this answers is not in the document]
Comment (peterhuffer) [reply catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitor.java]: Accident :)

**Comment by peterhuffer** [comment]
build now

**Comment by kcover** [comment]
Successful Hero! 🎊

**Comment by peterhuffer** [comment]
build now

**Pull request merged**